plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/ServicePortDialog.java

```diff
@@ -140,26 +141,23 @@ public IStatus validate(Object value) {
 	}
 	class PodPortValidator implements IValidator{
 		
-		private final int MAXLENGTH = 63;
-		private final Pattern REGEXP = Pattern.compile("[a-z0-9]([a-z0-9-]*[a-z0-9])*");
-
-		private final IStatus ERROR = ValidationStatus.error("The target port must be at most 15 characters, matching regex [a-z0-9]([a-z0-9-]*[a-z0-9])*, and hyphens cannot be adjacent to other hyphens): e.g. \"http\"");
+		private final IStatus ERROR = ValidationStatus.error("The target Pod port must be a number in range 0-65535");
 		
 		@Override
 		public IStatus validate(Object value) {
+			// port cannot be empty
 			if(StringUtils.isEmpty(value)) {
 				return ERROR;
 			}
-			String newPort = (String) value;
-			if(!newPort.equals(podPort)) {
-				if(newPort.length() > MAXLENGTH || !REGEXP.matcher(newPort).matches()) {
+			try {
+				final long portNumber = Long.valueOf(value.toString());
+				// validate range
+				if(portNumber < 0 || portNumber > 65535) {
```

Comment:
we still only validate to numbers? Shouldn't we keep the old validator but change the error-message? ports on the pod can be numbers or alphanum names.